### PR TITLE
Add kernel 4.4 configuration file

### PR DESCRIPTION
**Pull request by ghost**
Signed-off-by: Francois Berder Francois.Berder@imgtec.com

### CODE REVIEW

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

File: config-4.4-all [hunk 1]
```diff
@@ -612,19 +652,19 @@ CONFIG_USB_SERIAL_SIERRAWIRELESS=m
...
 CONFIG_USB_SUPPORT=y
-# CONFIG_USB_ULPI_BUS is not set
 CONFIG_USB_USBNET=m
 CONFIG_USB_U_SERIAL=m
 CONFIG_USELIB=y
 CONFIG_USER_NS=y
 CONFIG_USE_GENERIC_EARLY_PRINTK_8250=y
 CONFIG_USE_OF=y
 CONFIG_UTS_NS=y
+# CONFIG_VGA_CONSOLE is not set
```
Comment (abhijit-mahajani-imgtec) [inline config-4.4-all:632]: Why?
This is removing https://github.com/CreatorDev/ci/commit/fb9d84785a9e17e339f1f542026a2f157512ff68

**Comment by nikhil-zinjurde-imgtec** [comment]
The build-all CI job failed with this error

```
*
* Console display driver support
*
VGA text console (VGA_CONSOLE) [Y/n/?] (NEW) aborted!

Console input/output is redirected. Run 'make oldconfig' to update configuration.

make[8]: *** [silentoldconfig] Error 1
make[7]: *** [silentoldconfig] Error 2

The present kernel configuration has modules disabled.
Type 'make config' and enable loadable module support.
Then build a kernel with module support enabled.

make[6]: *** [modules] Error 1
```

File: Makefile [hunk 2]
```diff
@@ -48,7 +48,7 @@ $(DIR__OPENWRT)/version:
 .PHONY: build_openwrt
 build_openwrt: $(DIR__OPENWRT)/.config $(DIR__OPENWRT)/version
 ifneq (_,_$(findstring all,$P))
-	$(MAKE) $(SUBMAKEFLAGS) -C $(DIR__OPENWRT) IGNORE_ERRORS=1 -j$(J)
+	$(MAKE) $(SUBMAKEFLAGS) -C $(DIR__OPENWRT) IGNORE_ERRORS=y -j$(J)
```
Comment (abhijit-mahajani-imgtec) [inline Makefile]: Since all the packages generally get marked as modules in case of build-all i would change to IGNORE_ERRORS=m

**Comment by abhijit-mahajani-imgtec** [comment]
0001 glib patch needs to be updated.

File: Makefile [hunk 3]
```diff
@@ -23,6 +23,8 @@ $(DIR__CI)/patched:
 ifneq (_,_$(findstring all,$P))
 	cd $(DIR__OPENWRT)/feeds/packages; patch -p1 < $(DIR__CI)/0001-glib2-make-libiconv-dependent-on-ICONV_FULL-variable.patch; \
 	patch -p1 < $(DIR__CI)/0001-node-host-turn-off-verbose.patch; \
+	cd $(DIR__OPENWRT)/; \
+	patch -p1 < $(DIR__CI)/0001-package-Remove-zram-kernel-module.patch; \
```
Comment (abhijit-mahajani-imgtec) [inline Makefile]: filename is 0004-package-Remove-zram-kernel-module.patch but you are patching
 0001-package-Remove-zram-kernel-module.patch

File: Makefile [hunk 4]
```diff
@@ -23,6 +23,8 @@ $(DIR__CI)/patched:
 ifneq (_,_$(findstring all,$P))
 	cd $(DIR__OPENWRT)/feeds/packages; patch -p1 < $(DIR__CI)/0001-glib2-make-libiconv-dependent-on-ICONV_FULL-variable.patch; \
 	patch -p1 < $(DIR__CI)/0001-node-host-turn-off-verbose.patch; \
```
Comment (nikhil-zinjurde-imgtec) [inline Makefile]: This patch can now be removed as it has been verified that build can progress without this patch

**Review by abhijit-mahajani-imgtec — COMMENTED** [review]
Can't we just disable the kmod-zram after you load the creator-platform-all-cascoda.config, instead of patching the openwrt makefiles?
something like this?
echo "PACKAGE_kmod-zram=n" >> ./.config;

**Review by abhijit-mahajani-imgtec — COMMENTED** [review]
echo "PACKAGE_kmod-zram=n" >> ./.config;
not able to change the value which is
PACKAGE_kmod-zram=m

**Review by abhijit-mahajani-imgtec — APPROVED** [review]

**Review by abhijit-mahajani-imgtec — APPROVED** [review]

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Review by abhijit-mahajani-imgtec — APPROVED** [review]

**Review by abhijit-mahajani-imgtec — APPROVED** [review]

**Review by abhijit-mahajani-imgtec — APPROVED** [review]

**Review by abhijit-mahajani-imgtec — APPROVED** [review]

**Review by abhijit-mahajani-imgtec — APPROVED** [review]

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Pull request merged**